### DISCUSSION TITLE
How to set up SPSync + eq + some visualization with hardware DAC

### DISCUSSION THREAD

**Discussion by ghost** [mikebrady/shairport-sync#1907] [Q&A] [answered]
Hi, 
First of all immense kudos to Mike and the ones before him for what SPSync is today!

I've got an RPi4B with the IQAudIODAC hat on it, working fine just by making the card the default output in ALSA. Running SPSync 4.3.4 (I see nothing wrong playing to shairport and my mac.

First, how to use a graphic or parametric EQ with SPSync? (other than feeding a file to the convolution filter... that sure should be good but too advanced for me, I want controls I can play with).

The second is less important of course, but I wanted to get the audio stream so I can run (eg) cava. Is there a way for this?

Thanks!!!

Pedro

(i18 / 4.3.4-1-g910264e5-AirPlay2-smi10-alac-OpenSSL-Avahi-ALSA-pa-pw-stdout-pipe-soxr-convolution-metadata-dbus-mpris-sysconfdir:/etc)

**Comment by mikebrady** [comment, marked as answer]
Thanks for the kind words.

Instead of using the `alsa` backend, you would use the `pipe` (BTW this refers to a unix pipe, not PipeWire!) or the `stdout` backends to get the decoded PCM audio stream and then feed it in to whatever further processing you wish.

**Reply by ghost** [reply]
Thanks @mikebrady - will look into that!
